specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/Patches.json

```diff
@@ -0,0 +1,542 @@
+            "$ref": "./examples/Patches_Skip_Configure.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/builders/{builderName}/patches/{patchName}/apply": {
+      "post": {
+        "tags": [
+          "Patches"
+        ],
+        "operationId": "Patches_Apply",
+        "description": "Apply a PatchResource with patch name.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./Builders.json#/parameters/BuilderNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PatchNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Patching operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            }
+          },
+          "202": {
+            "description": "Patching operation has been started.",
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            },
+            "headers": {
+              "Azure-AsyncOperation": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Patches_Apply_0": {
+            "$ref": "./examples/Patches_Apply.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PatchResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PatchProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      },
+      "description": "The resource representation of a patch.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "PatchProvisioningState": {
+      "type": "string",
+      "description": "Resource instance provisioning state.",
+      "enum": [
+        "Succeeded",
```

Comment:
>Succeeded [](http://example.com/codeflow?start=9&length=9)

Why is there no non-terminal state declared, like 'Upating', 'Creating', 'Applying' or 'Downloading'?

Reply:
(Deleting is another popular one)

Reply:
since patch is a proxy resource of the "builder" resource, the resource provisioning is done when builder resource spins up. Patch resources won't be created until builder resource successfully provisioned the computing resource for applying patch object, hence no non-terminal state.

Reply:
Hmm. You're saying its possible to create the patch resource on a builder *before* provisioning the builder resource?

This sound suspiciously like it doesn't match the normal ARM meaning for provisioningState. ARM relies on Provisioning state to understand when the patch resource has finished being created, after e.g. a PUT request. But if it doesn't really start being created straight away, ARM could either poll, waiting forever,... or just give up. Since you must be returning a 'terminal' state immediately?

It might be clearer to make this not a provisioningState property but a separate property, if its not intended for integration into ARM deployment templates...

Or just not to have it at all.

Reply:
Sorry for the confusion. here. I meant that the provisioning of builder resource includes the provisioning of the patch compute resource, which is a container app job. Therefore, the non-terminal status in builder should reflects the status of the patch compute resource (which is different from these metadata returned from these arm calls).

Meanwhile, the control plane periodically retrieves the patch detection result from data plane. But those will be added into the db only if the builder resource provisioning state is in a successful state.

I do apologize for all these naming confusions. Please let me know if I could explain it more clearly!